assets/javascripts/lib/uploadHandler.js.es6

```diff
@@ -0,0 +1,48 @@
+import { isImage } from "discourse/lib/uploads";
```

Comment:
We should always be using Ember's `Promise` because it is ["integrated"](https://discuss.emberjs.com/t/readers-questions-why-does-ember-still-use-rsvp/14736) with the runloop.

```suggestion
import { isImage } from "discourse/lib/uploads";
import { Promise } from "rsvp";
```